Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -2084,14 +2084,16 @@ public void selectBlockers(Ability source, Game game, UUID defendingPlayerId) {
         if (skipButtonActivated && !canStopOnAny && !canStopOnZero) {
             return;
         }
+        java.util.List<UUID> possibleBlockers = game.getBattlefield().getActivePermanents(filter, playerId, game).stream()
+                .map(p -> p.getId())
+                .collect(Collectors.toList());
+        // Skip prompt to select blockers if player has none
+        if (possibleBlockers.isEmpty()) return;
```

Comment:
Well, it’s possible to make GUI settings to control in default mode too (without skip enabled):

// auto-skip select dialog by default (even without enabled skip button)
if (!skipButtonActivated && !canStopOnZero) {
  return;
}

Reply:
Just implemented the changes described above in ee7864e

Reply:
> if you have no creatures you don't have the means to assign any blockers anyway

That's single setting used in two places, search it by `isStopOnDeclareBlockersWithZeroPermanents`:
- for priority step before declare blocker;
- for declare blocker step.

The main idea behind zero blockers stop -- activate abilities before declare step.

The additional idea -- you can look at battlefield on declare blocker and find the reason of the miss blockers (e.g. no untapped creatures, can't block effects, block conditions, etc).

Reply:
I don't think you can do same research on priority after declare attacker. If it's ok for gameplay and players don't need declare blocker "pause" then you can keep new logic from PR -- skip empty blockers all the time.